docs/embedded/tutorials/migrate-abs-to-spe.md

```diff
@@ -0,0 +1,328 @@
+
+    var uploadSessionRequestBody = new CreateUploadSessionPostRequestBody()
+    {
+        AdditionalData = new Dictionary<string, object>
+        {
+            // Fail is set here, so it doesn't get upload again if it already exist
+            { "@microsoft.graph.conflictBehavior", "fail" }
+        }
+    };
+
+    var uploadSession = await _graphClient.Drives[_containerId]
+        .Items[parentFolderId]
+        .ItemWithPath(fileName)
+        .CreateUploadSession
+        .PostAsync(uploadSessionRequestBody);
+
+    // The stream is the same stream from the downloading the blob
+    var fileUploadTask = new LargeFileUploadTask<DriveItem>(uploadSession, stream, _maxChunkSize, _graphClient.RequestAdapter);
+        IProgress<long> progress = new Progress<long>(prog => Console.WriteLine($"Uploaded {fileName} {prog} bytes"));
+
+    // Check uploadResult.UploadSucceeded to see if it is successful
+    var uploadResult = await fileUploadTask.UploadAsync(progress);
+```
+
+## Handling Errors and Exceptions
+
+### Common Issues
+
+1. File already exists in destination
+    - This app checks to see if the file name exists in the destination before it uploads. If there is a file with the exact same name, it will not do the upload again. It will print to stdout a message that the file already exists. To fix it you can either delete the file from the destination or change the conflictBehavior to replace and not call `CheckIfItemExists` on upload.
+1. Passing a blob list
+    - The blob list is string that would be converted into Json value. The double quotes that wrap each item in the array would need to be escaped. The app checks if this value is in correct format, if it isn't it will print an error to stdout and exit the program.
+    - For example:
+        - Correct: `"[\""example1.txt\", \""example2.txt\", \""example3.txt\"]"`
+        - Incorrect: `"["example1.txt", "example2.txt", "example3.txt"]"`
+1. Not giving enough permission to access the ABS container
+    - The minimum permissions are Read and List
+1. Assigning the correct permissions to the SPE container
+    - The container exists because you can get lists of all the containers. However, when you try to do something to act on the container, it just exits the program.
+
+## Testing the Migration
+
+### Verification
+1. When the file is queued, it will print to stdout.
+1. If all blobs are uploaded successfully, it will print to stdout `Congratulations`
+1. If there are errors, it will output it to a file. The file name will be print to stdout. It will also, print out a command for incremental re-run.
+
+## Conclusion
+
+### Summary
+
+In this tutorial, we explored how to migrate content from ABS container to SPE container. By following the steps outlined, writing your own app to migrate content should be easy.
+
+To recap, we:
+1. Authenticated with ABS and Graph
+1. Retrieved the blob list from ABS container
+1. Uploaded the blob to the SPE container
```

Comment:
I am understanding from this tutorial the SPE container to be migrated to needs to be created prior to migrate, correct?